File: python/resistor-color-duo/resistor_color_duo.py
color_list = ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white']
value_list = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]

def value(colors):
    band_one, band_two, *_ = colors
    return color_code(band_one) * 10 + color_code(band_two)

def color_code(color):
    for index, c in enumerate(color_list):
        if color == c:
            return value_list[index]

File: java/resistor-color-duo/src/main/java/ResistorColorDuo.java
import java.util.Arrays;

class ResistorColorDuo {
    private final int COLOR_QUANTITY = 2;

    int value(String[] colors) {
        int sum = 0;
        int exponent = 0;
        for (int i = COLOR_QUANTITY - 1; i >= 0; i--) {
            sum += colorCode(colors[i]) * Math.pow(10, exponent);
            exponent++;
        }
        return sum;
    }

    int colorCode(String color) {
        return Arrays.asList(colors()).indexOf(color);
    }

    String[] colors() {
        return new String[] {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet",
                "grey", "white"};
    }
}
